res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,736 @@
+                        midi.sendShortMsg(i, 0x1C - 1 + samplerChan, value ?
+                            RED : OFF);
+                    }
+                }
+            }
+            break;
+    }
+};
+
+ReloopBeatmix24.SamplerPlay = function(value, group, control) {
+    var samplerRegEx = /\[Sampler(\d+)\]/;
+    var samplerChan = parseInt(samplerRegEx.exec(group)[1]);
+    if (samplerChan <= 8) { // We only handle 8 samplers (1 per pad)
+        var ledColor;
+        if (value) {
+            ledColor = VIOLET;
+        } else {
+            ledColor = engine.getValue(group, "track_samples") ? RED : OFF;
+        }
+
+        // We need to switch off pad lights before changing color otherwise
+        // VIOLET to RED transition does not work well. (???)
+        for (var i = 0x91; i <= 0x94; i++) {
+            // PAD1 Mode A
+            midi.sendShortMsg(i, 0x08 - 1 + samplerChan, OFF);
+            midi.sendShortMsg(i, 0x08 - 1 + samplerChan, ledColor);
+            // SHIFT+PAD1 Mode A
+            midi.sendShortMsg(i, 0x48 - 1 + samplerChan, OFF);
+            midi.sendShortMsg(i, 0x48 - 1 + samplerChan, ledColor);
+            if (samplerChan <= 4) { // Handle first 4 samplers in split mode
+                // PAD5 Mode A+B (sampler 1 in split mode)
+                midi.sendShortMsg(i, 0x14 - 1 + samplerChan, OFF);
+                midi.sendShortMsg(i, 0x14 - 1 + samplerChan, ledColor);
+                // SHIFT+PAD5 Mode A+B (sampler 1 in split mode)
+                midi.sendShortMsg(i, 0x1C - 1 + samplerChan, OFF);
+                midi.sendShortMsg(i, 0x1C - 1 + samplerChan, ledColor);
+            }
+        }
+    }
+};
+
+ReloopBeatmix24.loopDefined = function(value, group, control) {
+    var channelRegEx = /\[Channel(\d+)\]/;
+    var channelChan = parseInt(channelRegEx.exec(group)[1]);
+    if (channelChan <= 4) {
+        midi.sendShortMsg(0x90 + channelChan, 0x44, value < 0 ? OFF :
+            VIOLET);
+    }
+};
+
+ReloopBeatmix24.ChannelPlay = function(value, group, control) {
+    // Keep track of the playing state of each channel to avoid
+    // calling engine.getValue(group, "play") too often
+    if (value) {
+        channelPlaying[group] = true;
+    } else {
+        // Stop JogWheel blinking when we stop playing and resume virtual needle
+        if (JogBlinking[group]) {
+            engine.stopTimer(jogWheelTimers[group]);
+            delete jogWheelTimers[group];
+            JogBlinking[group] = false;
+            var channelRegEx = /\[Channel(\d+)\]/;
+            var channelChan = parseInt(channelRegEx.exec(group)[1]);
+            ReloopBeatmix24.AllJogLEDsToggle(0x90 + channelChan, OFF, 2);
+            engine.trigger(group, "playposition"); // lit up jog position led
+        }
+        channelPlaying[group] = false;
+    }
+};
+
+ReloopBeatmix24.JogLed = function(value, group, control) {
+
+    // time computation
+    var trackDuration = engine.getValue(group, "duration");
```

Comment:
Like the regexes, it may be a good idea to move this to a variable outside this function's scope that gets updated in the ChannelPlay function to avoid calling engine.getValue() on each update of playposition.